### PR TITLE
SLS-486 Don't evict pages that belongs to the next checkpoint

### PR DESCRIPTION
**Pull request by quchenhao**
Summarize the reason behind this change (this might be the problem you're solving, or the context around the request) and the solution you have chosen.

### CODE REVIEW

File: src/include/btree_inline.h [hunk 1]
```diff
@@ -1963,6 +1965,16 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_splitp)
         return (false);
     }
 
+    /* Don't evict the disaggregated page that should belong to the next checkpoint. */
+    if (modified && F_ISSET(btree, WT_BTREE_DISAGGREGATED) &&
+      btree->checkpoint_gen == __wt_gen(session, WT_GEN_CHECKPOINT))
+        /*
+         * TODO: we only know that the btree has been visited by the latest checkpoint. But we don't
```
Comment (pmacko86) [inline src/include/btree_inline.h:1972]: Does this mean that we cannot evict a page from a table in between checkpoints? I'm wondering if there is something that we could do based on `conn->disaggregated_storage.global_checkpoint_id`, from which you know whether the next checkpoint has been opened.

Comment (quchenhao) [reply src/include/btree_inline.h:1972]: Yes, currently we cannot evict. I was thinking about that yesterday and I think we need to remember the last completed checkpoint id.

Comment (quchenhao) [reply src/include/btree_inline.h]: I think I have a better idea. Updated the code.

**Review by pmacko86 — APPROVED** [review]
LGTM. I like how elegant this change is—thanks!

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**